x/wasm/keeper/proposal_handler.go

```diff
@@ -64,11 +65,15 @@ func handleStoreCodeProposal(ctx sdk.Context, k types.ContractOpsKeeper, p types
 	if err != nil {
 		return sdkerrors.Wrap(err, "run as address")
 	}
-	codeID, _, err := k.Create(ctx, runAsAddr, p.WASMByteCode, p.InstantiatePermission)
+	codeID, checksum, err := k.Create(ctx, runAsAddr, p.WASMByteCode, p.InstantiatePermission)
 	if err != nil {
 		return err
 	}
 
+	if !bytes.Equal(checksum, p.CodeHash) {
```

Comment:
There may be in flight proposals without this data. It would be good to not enforce the check when p.CodeHash is empty

Reply:
I think you got me wrong. `if !bytes.Equal(checksum, p.CodeHash) ` makes a lot of sense and ensures consistency of the proposal. Please add it back but with a `if p.CodeHash != nil && !bytes.Equal(checksum, p.CodeHash)` so that it works for old in flight proposals (without the new fields) , too.